drivers/i2c/i2c_gpio.c

```diff
@@ -24,41 +27,51 @@
 #include <errno.h>
 #include <drivers/gpio.h>
 #include <drivers/i2c.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(i2c_gpio);
+
+#include "i2c-priv.h"
 #include "i2c_bitbang.h"
 
 /* Driver config */
 struct i2c_gpio_config {
-	char *gpio_name;
-	u8_t scl_pin;
-	u8_t sda_pin;
+	const char *scl_gpio_name;
+	const char *sda_gpio_name;
+	gpio_pin_t scl_pin;
+	gpio_pin_t sda_pin;
+	gpio_dt_flags_t scl_flags;
+	gpio_dt_flags_t sda_flags;
+	u32_t bitrate;
 };
 
 /* Driver instance data */
 struct i2c_gpio_context {
 	struct i2c_bitbang bitbang;	/* Bit-bang library data */
-	struct device *gpio;		/* GPIO used for I2C lines */
-	u8_t scl_pin;			/* Pin on gpio used for SCL line */
-	u8_t sda_pin;			/* Pin on gpio used for SDA line */
+	struct device *scl_gpio;	/* GPIO used for I2C SCL line */
+	struct device *sda_gpio;	/* GPIO used for I2C SDA line */
+	gpio_pin_t scl_pin;		/* Pin on gpio used for SCL line */
+	gpio_pin_t sda_pin;		/* Pin on gpio used for SDA line */
 };
 
 static void i2c_gpio_set_scl(void *io_context, int state)
```

Comment:
Not in scope for this revision, but it seems odd to be passing this as a `void *` when the implementation requires that it be a `struct i2c_gpio_context`.  If that could be cleaned up maybe we could avoid duplicating the pin storage in two structures.

Reply:
Yes, agreed. I plan to rework that in a separate PR.